### DISCUSSION TITLE
Release the GIL for long-running functions?

### DISCUSSION THREAD

**Discussion by TheRealQuantam** [python-pillow/Pillow#6410] [Q&A] [answered]
Would it be possible to release the GIL for long-running functions?

e.g WebPAnimEncoder.add (especially at high quality settings).

**Comment by radarhere** [comment]
Could you provide some more information? What negative situation are you experiencing that caused you to figure out that Pillow wasn't releasing?

**Reply by TheRealQuantam** [reply]
The test was really simple: I just spawned 2 threads and had both call save for 2 different WebP Images (though I don't know if it matters, but both were Image.copy's of the same image). It took about 2x as long as a single thread saving 1 image. That and CPU usage never went above 12.5%.

**Comment by radarhere** [comment]
You may be interested in https://github.com/python-pillow/Pillow/pull/7782, which releases the GIL while calling `WebPAnimDecoderGetNext`

**Comment by radarhere** [comment]
You may be interested in #8199, where Pillow 11.0.0 will work with the experimental free-threaded mode of Python 3.13.

**Comment by radarhere** [comment, marked as answer]
#8386 will finally release the GIL for `WebPAnimEncoderAdd`
